src/main/java/de/komoot/photon/query/PhotonRequestFactory.java

```diff
@@ -0,0 +1,120 @@
+package de.komoot.photon.query;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.Point;
+import com.vividsolutions.jts.geom.PrecisionModel;
+import spark.QueryParamsMap;
+import spark.Request;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * A factory that creates a {@link PhotonRequest} from a {@link Request web request}
+ * Created by Sachin Dole on 2/12/2015.
+ */
+public class PhotonRequestFactory {
+
+    private final LanguageChecker languageChecker;
+
+    public PhotonRequestFactory(Set<String> supportedLanguages) {
+        this.languageChecker = new LanguageChecker(supportedLanguages);
+    }
+
+    public <R extends PhotonRequest> R create(Request webRequest) throws BadRequestException {
+        String language = webRequest.queryParams("lang");
+        language = language == null ? "en" : language;
+        languageChecker.apply(language);
+        String query = webRequest.queryParams("q");
+        if (query == null) throw new BadRequestException(400, "missing search term 'q': /?q=berlin");
+        Integer limit;
+        try {
+            limit = Integer.valueOf(webRequest.queryParams("limit"));
+        } catch (NumberFormatException e) {
+            limit = 15;
+        }
+        Point locationForBias = null;
+        try {
+            Double lon = Double.valueOf(webRequest.queryParams("lon"));
+            Double lat = Double.valueOf(webRequest.queryParams("lat"));
+            locationForBias = new GeometryFactory(new PrecisionModel(), 4326).createPoint(new Coordinate(lon, lat));
```

Comment:
there is no need to create a geometry factory for every request, i can be a static final field